src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -1205,6 +1205,30 @@ class QueryAnalyzer
 
     static std::string rewriteAggregateFunctionNameIfNeeded(const std::string & aggregate_function_name, const ContextPtr & context);
 
+    static std::optional<JoinTableSide> getColumnSideFromJoinTree(QueryTreeNodePtr & resolved_identifier, const JoinNode & join_node)
+    {
+        const auto * column_src = resolved_identifier->as<ColumnNode &>().getColumnSource().get();
```

Comment:
Yes, it can be `SELECT ta.t.* FROM ta FULL JOIN tb USING (t.t.t);` where `ta.t` is nested.
 
I added handling to this function in this PR , but will add test in https://github.com/ClickHouse/ClickHouse/pull/49703 , because without tat fix it will be fail in other place